tests/security/swtpm/swtpm_verify.pm

```diff
@@ -17,19 +17,14 @@ use version_utils 'is_sle';
 sub run {
     select_serial_terminal if !(get_var('MACHINE') =~ /RPi4/);
     if (is_sle('=15-SP6')) {
-        record_soft_failure('SKIPPING TEST - poo#179729');
+        record_soft_failure('SKIPPING TEST; poo#179729 bsc#1242989');
         return;
     }
-    my $vm_type = 'legacy';
-    $vm_type = 'uefi' if get_var('UEFI');
+    my $vm_type = get_var('UEFI') ? 'uefi' : 'legacy';
     # aarch64 does not support tpm1.2
     my @swtpm_versions = qw(swtpm_2);
-    # do not test TPM 1.2 on aarch64 or SLE >= 15-SP6 if FIPS is enabled
-    if (!is_aarch64) {
-        if (is_sle('<=15-SP5') || (is_sle('>=15-SP6') && (!get_var('FIPS_ENABLED') || check_var('FIPS_ENABLED', '0')))) {
-            push @swtpm_versions, qw(swtpm_1);
-        }
-    }
+    # Do not test TPM 1.2 on aarch64 or SLE >=15-SP6 if FIPS is enabled
+    push @swtpm_versions, 'swtpm_1' if (!is_aarch64) && (is_sle('<=15-SP5') || (is_sle('>=15-SP6') && (!get_var('FIPS_ENABLED') || check_var('FIPS_ENABLED', '0'))));
```

Comment:
this logic statement is a bit hard to grasp, would be better to simplify it.
Seems you are checking for negative conditions, so I'd group them and maybe assign the condition to an intermediate bool variable with a meaningful name

Reply:
yes, the first condition might be redundant but I find the rest very readable :) WDYT ?

Reply:
It's a matter of taste, granted this is more maintainable. I've pushed a similar change now, running new VRs.